src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -147,7 +147,28 @@ export default function Inbox ({ history: { push } }) {
   const refundAllDeclinedTransactions = useRefundAllDeclinedTransactions()
   const [counterpartyNotFound, setCounterpartyNotFound] = useState(true)
   const [isDeclinedTransactionModalVisible, setIsDeclinedTransactionModalVisible] = useState(false)
-  const [modalTransaction, setModalTransaction] = useState(null)
+  const [confirmationModalProperties, setConfirmationModalProperties] = useState({
+    shouldDisplay: false,
+    transactions: [],
+    action: '',
+    hasConfirmed: false
+  })
+
+  const resetDefaultModalTransaction = () => {
+    setConfirmationModalProperties({
+      shouldDisplay: false,
+      transactions: [],
+      action: '',
+      hasConfirmed: false
+    })
+  }
+
+  const setNewModalTransactionProperties = (newProperties = {}) => {
+    if (isEmpty(newProperties)) resetDefaultModalTransaction()
+    else setConfirmationModalProperties(newProperties)
+  }
+
+  const clearHighlightedTransaction = timeout => setTimeout(() => resetDefaultModalTransaction(), timeout)
```

Comment:
this name seems inaccurate to me. Doesn't this close the modal?

Reply:
We had a prop in the modals called `handleClose` before, which would call this same state hook and set it to null.  Since we're now using that same state hook to update the actioned-transaction row itself, I decided to only pass in this setState hook once instead of twice (once explicitly for `handleClose `and the other for updating the state with the params).